src/services/services.ts

```diff
@@ -4639,7 +4639,16 @@ namespace ts {
             // to jump to the implementation directly.
             if (symbol.flags & SymbolFlags.Alias) {
                 const declaration = symbol.declarations[0];
-                if (node.kind === SyntaxKind.Identifier && node.parent === declaration) {
+
+                // Go to the original declaration for cases:
+                //
+                //   (1) when the aliased symbol was declared in the location(parent).
+                //   (2) when the aliased symbol is originating from a named import.
+                //
+                if (node.kind === SyntaxKind.Identifier &&

```

Comment:
I'm curious if we need this condition at all. It used to be necessary to distinguish when we to GotoDef on use-site of import vs its declaration. If we are saying that we always want to go to the ultimate location I think that condition is unnecessary.